web/src/Backend.js

```diff
@@ -1,129 +1,175 @@
 import * as Setting from "./Setting";
 
 export function getAdapters() {
-    return fetch(`${Setting.ServerUrl}/api/get-adapters`, {
-        method: "GET",
-        credentials: "include"
-    }).then(res => res.json());
+  return fetch(`${Setting.ServerUrl}/api/get-adapters`, {
+    method: "GET",
+    credentials: "include",
+  })
+    .then(res => res.json())
+    .catch(err => console.error(err));
 }
 
 export function getAdapter(adapterId) {
-    return fetch(`${Setting.ServerUrl}/api/get-adapter?id=${adapterId}`, {
-        method: "GET",
-        credentials: "include"
-    }).then(res => res.json());
+  return fetch(`${Setting.ServerUrl}/api/get-adapter?id=${adapterId}`, {
+    method: "GET",
+    credentials: "include",
+  })
+    .then(res => res.json())
+    .catch(err => console.error(err));
 }
 
 export function updateAdapters(adapters) {
-    return fetch(`${Setting.ServerUrl}/api/update-adapters`, {
-        method: 'POST',
-        credentials: 'include',
-        body: JSON.stringify(adapters),
-    }).then(res => res.json());
+  return fetch(`${Setting.ServerUrl}/api/update-adapters`, {
+    method: "POST",
+    credentials: "include",
+    body: JSON.stringify(adapters),
+  })
+    .then(res => res.json())
+    .catch(err => console.error(err));
 }
 
 export function updateAdapter(adapter) {
-    return fetch(`${Setting.ServerUrl}/api/update-adapter`, {
-        method: 'POST',
-        credentials: 'include',
-        body: JSON.stringify(adapter),
-    }).then(res => res.json());
+  return fetch(`${Setting.ServerUrl}/api/update-adapter`, {
+    method: "POST",
+    credentials: "include",
+    body: JSON.stringify(adapter),
+  })
+    .then(res => res.json())
+    .catch(err => console.error(err));
 }
 
 export function testAdapterConnection(adapter) {
-    return fetch(`${Setting.ServerUrl}/api/test-adapter-connection`, {
-        method: 'POST',
-        credentials: 'include',
-        body: JSON.stringify(adapter),
-    }).then(res => res.json());
+  return fetch(`${Setting.ServerUrl}/api/test-adapter-connection`, {
+    method: "POST",
+    credentials: "include",
+    body: JSON.stringify(adapter),
+  })
+    .then(res => res.json())
+    .catch(err => console.error(err));
 }
 
 export function getAdapterPolicies(adapterId) {

```

Comment:
As same catch statement is used everywhere so I think you should create an observe function to catch them instead of writing them multiple time.

Reply:
same try statement too is written .
Also, it is the okay to use try-catch in such a manner to give a better understanding to other coders.

Reply:
But we need to think of scalability also.